pkg/skaffold/server/proto/skaffold.proto

```diff
@@ -92,4 +92,11 @@ service SkaffoldService {
       body: "*"
     };
   }
+
+  rpc Build(google.protobuf.Empty) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/v1/build"
```

Comment:
What is build going to do? will it build, test and deploy? or just build? Will it run a complete build or build a specific artifact? Can the build api provide this fine level granularity?
The reason i ask is may be the Trigger `chan bool` might not be sufficient. 
If we want to force build a specific artifact, then the we need a richer chan struct.
Something to think about for future.

Reply:
the behavior is exactly the same as if you were to make a change to a file and save it while running `skaffold dev`. the only difference here is that instead of skaffold watching your files for changes and then triggering a cycle, now the user will trigger the cycle manually. the next step after this PR is to implement a deploy trigger, which will override the dev loop's deploy in favor of a manual trigger for that as well. but for now, deploys happen whenever a rebuild finishes successfully, as usual.

it doesn't make sense to have anything other than `chan bool` here because the watcher will still compute the changes from the previous build and determine which artifacts to build.

Reply:
got it. This comment was also before your explanation.